File: lector_grafo.py
# lector_grafo.py
# Lee el archivo guategrafo.txt y carga el grafo

def leer_archivo(ruta, grafo):
    arcos = 0
    try:
        with open(ruta, "r", encoding="utf-8") as f:
            for num, linea in enumerate(f, 1):
                linea = linea.strip()
                if not linea:
                    continue
                partes = linea.split()
                if len(partes) < 3:
                    print(f"  [!] Linea {num} ignorada (formato incorrecto): {linea}")
                    continue
                try:
                    km = int(partes[2])
                    if km <= 0:
                        raise ValueError
                    grafo.agregar_arco(partes[0], partes[1], km)
                    arcos += 1
                except ValueError:
                    print(f"  [!] Linea {num} KM invalido: {linea}")
    except FileNotFoundError:
        print(f"  Error: No se encontro el archivo '{ruta}'")
        return -1
    return arcos

File: LectorGrafo.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class LectorGrafo {

    public static int leerArchivo(String ruta, Grafo grafo) {
        int arcos = 0;
        try (BufferedReader br = new BufferedReader(new FileReader(ruta))) {
            String linea;
            int numLinea = 0;
            while ((linea = br.readLine()) != null) {
                numLinea++;
                linea = linea.trim();
                if (linea.isEmpty()) continue;
                String[] partes = linea.split("\\s+");
                if (partes.length < 3) {
                    System.out.println("  [!] Linea " + numLinea + " ignorada: " + linea);
                    continue;
                }
                try {
                    int km = Integer.parseInt(partes[2]);
                    if (km <= 0) throw new NumberFormatException();
                    grafo.agregarArco(partes[0], partes[1], km);
                    arcos++;
                } catch (NumberFormatException e) {
                    System.out.println("  [!] Linea " + numLinea + " KM invalido: " + linea);
                }
            }
        } catch (IOException e) {
            System.out.println("  Error al leer archivo: " + e.getMessage());
            return -1;
        }
        return arcos;
    }
}
